src/main/java/org/isf/patient/rest/PatientController.java

```diff
@@ -105,40 +104,40 @@ public PatientController(PatientBrowserManager patientManager, PatientMapper pat
 		return ResponseEntity.ok(patientMapper.map2DTO(patient));
 	}
 
-
-	@GetMapping(value = "/patients/search", produces = MediaType.APPLICATION_JSON_VALUE)
-	public ResponseEntity<PatientDTO> searchPatient(
-			@RequestParam(value="name", defaultValue="") String name,
-			@RequestParam(value="code", required=false) Integer code) throws OHServiceException {
-        logger.info("Search patient name:" + name + " code:"  +  code);
-		Patient patient = null;
-		if(code != null) {
-            patient = patientManager.getPatient(code);
-		}else if (!name.equals("")) {
-            patient = patientManager.getPatient(name);
-		}
-        if (patient == null) {
-            return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
-        }
-        return ResponseEntity.ok(patientMapper.map2DTO(patient));
-	}
-
     @PostMapping(value = "/patients/search", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<List<PatientDTO>> searchPatient(
             @RequestParam(value="firstName", defaultValue="", required = false) String firstName,
-            @RequestParam(value="secondName", defaultValue="", required = false) String secondName
+            @RequestParam(value="secondName", defaultValue="", required = false) String secondName,
+            @RequestParam(value="birthDate", defaultValue="", required = false) String birthDate,
+            @RequestParam(value="address", defaultValue="", required = false) String address
     ) throws OHServiceException {
 
 
         List<Patient> patientList = null;
 
-        Map<String, String> params = new HashMap<String, String>();
+        Map<String, Object> params = new HashMap<String, Object>();
```

Comment:
Minimum JDK is JDK 8 which doesn't require data types for new statements; it gets them from the variable declaration.